src/main/groovy/com/getkeepsafe/dexcount/sdkresolver/SdkResolver.groovy

```diff
@@ -0,0 +1,122 @@
+package com.getkeepsafe.dexcount.sdkresolver
+
+import org.gradle.api.Project
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.tasks.StopExecutionException
+
+import static com.android.SdkConstants.ANDROID_HOME_ENV
+import static com.android.SdkConstants.FN_LOCAL_PROPERTIES
+import static com.android.SdkConstants.PLATFORM_WINDOWS
+import static com.android.SdkConstants.SDK_DIR_PROPERTY
+import static com.android.SdkConstants.currentPlatform
+
+/**
+ * This was lifted and modified from Jake Wharton's SDK Manager gradle plugin:
+ * https://github.com/JakeWharton/sdk-manager-plugin
+ *
+ * Copyright 2014 Jake Wharton
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class SdkResolver {
+    static File resolve(Project project) {
+        boolean isWindows = currentPlatform() == PLATFORM_WINDOWS
+        return new SdkResolver(project, new System.Real(), isWindows).resolve()
+    }
+
+    final Logger log = Logging.getLogger SdkResolver
+    final Project project
+    final System system
+    final File userHome
+    final File userAndroid
+    final File localProperties
+    final boolean isWindows
+
+    SdkResolver(Project project, System system, boolean isWindows) {
+        this.project = project
+        this.system = system
+        this.isWindows = isWindows
+
+        userHome = new File(system.property('user.home'))
+        userAndroid = new File(userHome, '.android-sdk')
```

Comment:
I notice that, on my system at least, there is no `.android-sdk` directory.  Does this exist for you?  What installed it?

Also, Android Studio stores its SDK elsewhere; can we account for that here?